enhancements/installer/assisted-installer.md

```diff
@@ -0,0 +1,365 @@
+vendor.
+
+## Motivation
+
+The target user is someone wanting to deploy OpenShift, especially on
+bare metal, with as little up-front infrastructure dependencies as
+possible. This person has access to servers and wants to run workloads
+quickly. They do not necessarily have the administrative privileges to
+create private VLANs, configure DHCP/PXE servers, or manage other
+aspects of the infrastructure surrounding the hardware where the
+cluster will run. If they do have the required privileges, they may
+not want to delegate them to the OpenShift installer for an IPI
+installation, preferring instead to use their existing tools and
+processes for some or all of that configuration. They are willing to
+accept that the cluster they build may not have all of the same
+features as an IPI cluster on day 1, but that by taking additional
+steps later they will be able to add more automatic infrastructure
+management and make it fully production-ready.
+
+### Goals
+
+- Make initial deployment of usable and supportable clusters simpler.
+- Move more infrastructure configuration from day 1 to day 2.
+- Support connected and disconnected on-premise deployments.
+- Support existing IPI features, especially for day 2 cluster
+  management and scale-out.
+
+### Non-Goals
+
+- Because the initial focus is for bare metal, this enhancement does
+  not exhaustively cover variations needed to offer similar features
+  on other platforms (such as changes to image formats, the way a host
+  boots, etc.). It is desirable to support those platforms, but that
+  work will be described separately.
+- Replace the existing OpenShift installer.
+- Describe how these workflows would work for multi-cluster
+  deployments managed with Hive or ACM.
+
+## Proposal
+
+There are several separate changes to enable the assisted installer
+workflows, including a GUI front-end for the installer, a cloud-based
+orchestration service, and changes to the installer and bootstrapping
+process.
+
+### Connected Deployments
+
+The process starts when the user goes to an "assisted installer"
+application running on `cloud.redhat.com`, enters details needed by
+the installer (OpenShift version, ssh keys, proxy settings, etc.), and
+then downloads a live RHCOS ISO image with the software and settings
+they need to complete the installation locally.
+
+The user then boots the live ISO on each host they want to be part of
+the cluster (control plane and workers). They can do this by hand
+using thumb drives, by attaching the ISO using virtual media support
+in the BMC of the host, or any other way they choose.
+
+When the ISO boots, it starts an agent that communicates with the REST
+API for the assisted installer service running on `cloud.redhat.com`
+to receive instructions. The agent registers the host with the
+service, using the user's pull secret embedded in the ISO's ignition
+to authenticate. The agent identifies itself based on the serial
```

Comment:
Can we be any more specific here?  Is there an API we can use to validate a pull secret is valid for a user, or will the assist service generate a new one per-host then store that in a DB?

Reply:
@avishayt or @romfreiman ?

Reply:
https://issues.redhat.com/browse/SDB-1117
The API will map the pull secret to the user's information

Reply:
Thanks - that link and the gitlab repo it refers to isn't public, is the idea that the auth code will be decoupled into the cloud connector so that the assisted service itself isn't coupled to that private API?

Reply:
My current direction is to implement authentication and authorization in a gateway, so that we can take the other code and more easily run it on prem.

Reply:
we dont have a complete design yet. Is the idea of enhancement to replace HLD docs? just to understand.

Reply:
Enhancement documents like this are required to make significant changes in OpenShift. They're a way for all of the various OpenShift teams to coordinate their work and understand the inter-team dependencies of that work. The level of detail seems to vary. Some teams use them exclusively. Other teams use them as a coordination point but supplement with details in other documents. I started by taking that second approach in this document because there are a lot of implementation details that won't matter for coordination.